### PR TITLE
Trying to fix mob diagonal dodging

### PR DESCRIPTION
**Pull request by Tk420634**
@OverDriveZ @Superlagg @TalkingCactus @CASEYGUNGEON

Not quite working, but if one of you would be so kind as to give it a sniff I'd appreciate it!
Should add a sound when they dodge, and a chance to smack into walls like a goober, with a message saying they dodged.

## Pre-Merge Checklist
- [ ] You tested this on a local server.
- [ ] This code did not runtime during testing.
- [ ] You documented all of your changes.

## Changelog
:cl:
add: Added new things
add: Added more things
del: Removed old things
tweak: tweaked a few things
balance: rebalanced something
fix: fixed a few things
soundadd: added a new sound thingy
sounddel: removed an old sound thingy
imageadd: added some icons and images
imagedel: deleted some icons and images
spellcheck: fixed a few typos
code: changed some code
refactor: refactored some code
config: changed some config setting
admin: messed with admin stuff
server: something server ops should know
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 1]
```diff
@@ -730,12 +730,29 @@
...
-	var/ccdir = turn(move_direction,-45)
+	// Assuming we move towards the target, we want to swerve toward them to get closer
+	var/cdir = turn(move_direction,90)
+	var/ccdir = turn(move_direction,-90)
 	dodging = FALSE
-	. = Move(get_step(loc,pick(cdir,ccdir)))
-	if(!.)//Can't dodge there so we just carry on
+	var/next_step_dir = pick(cdir,ccdir)
+	// Check if the direction is diagonal
+	if(next_step_dir % 2 != 0) // Diagonal directions are odd numbers in BYOND
```
Comment (CASEYGUNGEON) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm:739]: after testing,
```
N == 1
S == 2
E == 4
W == 8

NE == 5
SE == 6
NW == 9
SW == 10
```
i might just try
```
var/diagonals = list( NORTHEAST, SOUTHEAST, NORTHWEST, SOUTHWEST )
if(next_step_dir in diagonals)```

Comment (CASEYGUNGEON) [reply code/modules/mob/living/simple_animal/hostile/hostile.dm:739]: ooh oh oh oh i know what you can do if you want your code to be less readable!
```
var/logresult = log(next_step_dir, 2)
if(round(logresult) != logresult)
```

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 2]
```diff
@@ -730,12 +730,29 @@
...
+			playsound(loc, 'sound/effects/rustle3.ogg', 50, 1, -1)
+		} else {
+			// Apply stamina damage if the mob tried to dodge into a wall
+			adjustStaminaLoss(10)
+			visible_message("<span class='notice'>[src] tries to dodge but hits a wall!</span>")
+			playsound(loc, 'sound/effects/hit_punch.ogg', 50, 1, -1) // Play a punch sound
+		}
+	}
+	if(!.) // Can't dodge there so we just carry on
 		. =  Move(moving_to,move_direction)
```
Comment (CASEYGUNGEON) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm:756]: are we only running logic and playing audiovisuals when they dodge diagonally?

Comment (Tk420634) [reply code/modules/mob/living/simple_animal/hostile/hostile.dm:756]: Naw, the idea here is if they dodge at all (which should just be NESW) then they go 'haha i dodged', but if they happen to dodge into a wall they go 'oof' and take some damage.

Comment (CASEYGUNGEON) [reply code/modules/mob/living/simple_animal/hostile/hostile.dm:756]: ok i went and looked at the entire .dm to figure out why 756 calls `Move`. code is wack

**Pull request closed without merging**